utils/filecache/interfaces.go

```diff
@@ -0,0 +1,59 @@
+package filecache
```

Comment:
could file cache be a submodule of a top module being either filesystem or cache